src/App.tsx

```diff
@@ -1,24 +1,206 @@
+  const [isDisableInput, setIsDisableInput] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const completedTodos = todos.every((todo) => todo.completed);
+
+  const fetchData = async () => {
+    const todosFromServer = await getTodo(USER_ID);
+
+    try {
+      setTodo(todosFromServer);
+      setIsLoading(true);
+    } catch {
+      setError('Unable to load a todo');
+      setIsLoading(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const filterTodo = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterType) {
+        case FilterType.ALL:
+          return todo;
+
+        case FilterType.ACTIVE:
+          return !todo.completed;
+
+        case FilterType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterType]);
+
+  const handleAddTodo = async (title: string) => {
+    try {
+      setIsDisableInput(true);
+
+      const data = {
+        id: Math.floor(Math.random() * 100),
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      const newTodo = await postTodo(data);
+
+      setTodo(todo => [...todo, newTodo]);
+      setTempTodo({ ...data, id: 0 });
+    } catch {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setIsDisableInput(false);
+    }
+  };
+
+  const handleDeleteTodo = async (id: number) => {
+    try {
+      await deleteTodo(id);
+      const currentTodo = todos.filter((todo) => todo.id !== id);
+
+      setTodo(currentTodo);
+      setError('');
+    } catch {
+      setError('Unable to delete a todo');
+    }
+  };
+
+  const handleClearCompleted = async () => {
+    const completedMap = todos.filter(todo => todo.completed);
+
+    try {
+      completedMap.map(todo => deleteTodo(todo.id));
+
+      setTodo(todos.filter(todo => !todo.completed));
+    } catch {
+      setError('Unable to clear completed a todo');
+    }
+  };
+
+  const handleToggleAll = async () => {
+    try {
+      const updateCompleted = todos.map((todo) => {
+        if (todo.completed === completedTodos) {
+          updateTodo(todo.id, { completed: !completedTodos });
+        }
+
+        return { ...todo, completed: !completedTodos };
+      });
+
+      setTodo(updateCompleted);
+    } catch {
+      setError('Unable to update a todo');
+    }
+  };
+
+  const handleUpdateTodo = async (id: number, data: Partial<Todo>) => {
+    try {
+      await updateTodo(id, data);
+
+      setTodo(prev => prev.map(todo => {
+        if (todo.id === id) {
+          return { ...todo, ...data };
+        }
+
+        return todo;
+      }));
+    } catch {
+      setError('Unable to update a todo');
+    }
+  };
+
+  useEffect(() => {
+    if (error) {
+      const timeout = setTimeout(() => {
+        setError('');
+      }, 3000);
+
+      clearTimeout(timeout);
```

Comment:
move `clearTimeout(timeout)` to `return` statement for specifying a cleanup function.